lib/popup.js

```diff
@@ -1,106 +1,83 @@
+      return dayStats ? Object.keys(dayStats).sort((a, b) => dayStats[a] <= dayStats[b]).slice(0, 5) : []
+    },
+    timeSpent: function () {
+      if (this.dates[this.date]) {
+        if (this.stats[this.dates[this.date]]) {
+          if (this.stats[this.dates[this.date]][this.hostname]) {
+            this.noData = false
+            return formatTime(this.stats[this.dates[this.date]][this.hostname])
+          } else {
+            this.noData = true
+            return `You didn't visited this site ${this.date.toLowerCase()}!`
+          }
+        } else {
+          this.noData = false
+          return 'No data for this day'
+        }
+      } else {
+        this.noData = false
+        return 'Invalid date'
+      }
+    },
+    validDates: function () {
+      return Object.keys(this.dates).filter(date => this.stats && this.stats[this.dates[date]])
     }
-  })
-
-  function estimateTime () {
-    if (typeof results.hostNavigationStats[dates[displayDate]] === 'undefined' ||
-        typeof results.hostNavigationStats[dates[displayDate]][displayHostname] === 'undefined') {
-      timeSpentEl.textContent = 'No data'
-      return
+  },
+  methods: {
+    track: function () {
+      this.config.blacklist.push(this.hostname)
+      this.stats[this.dates.Today][this.hostname] = 1 // "Fake" time to be sure that it will show something after re-rendering
+      chrome.storage.local.set({ hostNavigationStats: this.stats, config: this.config })
     }
-
-    const elapsedTime = results.hostNavigationStats[dates[displayDate]][displayHostname]
-    updateTimer(elapsedTime)
   }
+})
 
-  // Get hostname of current tab
-  chrome.tabs.query({active: true, currentWindow: true}, tabs => {
-    const currentHostname = new URL(tabs[0].url).hostname
+chrome.storage.local.get({ hostNavigationStats: {}, config: {} }, res => {
+  res.hostNavigationStats[app.dates.Yesterday] = {
+    'twitter.com': 52,
+    'unixcorn.xyz': 42
+  }
+  app.stats = res.hostNavigationStats
+  app.config = res.config
+})
 
-    // Display five most visited websites
-    for (const hostname of sortedHostnames.slice(0, 5)) {
-      const networkItem = document.createElement('option')
-      networkItem.textContent = hostname
-      networkItem.value = hostname
-      // Select by default the current website
-      if (hostname === currentHostname) {
-        networkItem.selected = true
-      }
-      networkEl.appendChild(networkItem)
-    }
-    // Display the spent time for current tab
-    displayHostname = currentHostname
-    // Display if website is tracked or not
-    if (results.config.blacklist.includes(displayHostname)) {
-      nonMeasureDisplayEl.style.display = 'none'
-      estimateTime()
-      measureDisplayEl.style.display = 'block'
-    }
-  })
+chrome.tabs.query({active: true, currentWindow: true}, tabs => {
+  app.hostname = app.activeURL = new URL(tabs[0].url).hostname
 })
 
-function updateTimer (elapsedTime) {
+function formatTime (time) {
```

Comment:
You can also move this function to a `method` of the Vue app in order to be consistent